php/utils.php

```diff
@@ -544,11 +543,41 @@ function make_progress_bar( $message, $count, $interval = 100 ) {
 	return new \cli\progress\Bar( $message, $count, $interval );
 }
 
-function parse_url( $url ) {
-	$url_parts = \parse_url( $url );
+/**
+ * Helper function to use wp_parse_url when available or fall back to PHP's
+ * parse_url if not.
+ *
+ * Additionally, this adds 'http://' to the URL if no scheme was found.
+ *
+ * @param string $url           The URL to parse.
+ * @param int $component        Optional. The specific component to retrieve.
+ *                              Use one of the PHP predefined constants to
+ *                              specify which one. Defaults to -1 (= return
+ *                              all parts as an array).
+ * @param bool $auto_add_scheme Optional. Automatically add an http:// scheme if
+ *                              none was found. Defaults to true.
+ * @return mixed False on parse failure; Array of URL components on success;
+ *               When a specific component has been requested: null if the
+ *               component doesn't exist in the given URL; a string or - in the
+ *               case of PHP_URL_PORT - integer when it does. See parse_url()'s
+ *               return values.
+ */
+function parse_url( $url, $component = - 1, $auto_add_scheme = true ) {
+	if (
+		function_exists( 'wp_parse_url' )
+		&& (
+			-1 === $component
+			|| wp_version_compare( '4.7', '>=' )
+		)
+	) {
+		$url_parts = wp_parse_url( $url, $component );
+	} else {
+		// phpcs:ignore WordPress.WP.AlternativeFunctions.parse_url_parse_url -- Fallback.
+		$url_parts = \parse_url( $url, $component );
+	}
 
-	if ( ! isset( $url_parts['scheme'] ) ) {
-		$url_parts = parse_url( 'http://' . $url );
+	if ( $auto_add_scheme && ! isset( $url_parts['scheme'] ) ) {
+		$url_parts = WP_CLI\Utils\parse_url( 'http://' . $url, $component );
```

Comment:
:point_up: `namespace\`

Reply:
That was meant to be already there, pushed it once more.